src/containers/gui.jsx

```diff
@@ -13,6 +13,8 @@ import {
     getIsShowingProject
 } from '../reducers/project-state';
 import {setProjectTitle} from '../reducers/project-title';
+import {detectTutorialId} from '../lib/tutorial-from-url';
```

Comment:
is it possible to move this url detection to an HOC? Seems like that's the design pattern we're trying to follow, and moving towards.

I think we could do the same componentDidMount logic in the HOC, and it should just work, because gui watches the reducer values already.

What do you think? Partially, I'm thinking forward to our needing to mess with the url even more... seems like an HOC wrapping the whole gui at the top level would be good.